presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java

```diff
@@ -29,7 +31,12 @@
 
     void clear();
 
-    RowEncoder putNullValue(EncoderColumnHandle columnHandle);
+    RowEncoder putValueAt(EncoderColumnHandle columnHandle, ConnectorSession session, Block block, int position);
```

Comment:
Maybe `putValueFromBlock`? I do not know why but when I see `putValueAt` method which takes block I think it will be putting something into the block (though blocks are not mutable).

Reply:
Also can't you put default implementation already here?

Reply:
right, that could definitely be confusing

Reply:
I think so, would just be the same thing as what was in `KafkaPageSink#appendColumn`